### PR TITLE
Pinjin/jsonl convert

### PR DESCRIPTION
**Pull request by pj-ms**
- Add additional info support for all kinds of manifest
- "grounding" => "groundings" in coco json format
- image text matching accepts match value in float between 0 to 1
- proper support manifest => JSONL conversion

### CODE REVIEW

File: vision_datasets/commands/utils.py [hunk 1]
```diff
@@ -161,6 +164,19 @@ def convert_to_tsv(manifest: DatasetManifest, file_path):
             file_out.write(f'{img.id}\t{json.dumps(converted_labels, ensure_ascii=False)}\t{b64img}\n')
 
 
+def convert_to_jsonl(manifest: DatasetManifest, file_path: Union[str, pathlib.Path]):
```
Comment (msxingpeng) [inline vision_datasets/commands/utils.py]: unit test？

**Review by msxingpeng — APPROVED** [review]

**Review by shonohs — APPROVED** [review]

File: vision_datasets/commands/converter_to_line_oriented_format.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+def logging_prefix(dataset_name, version):
+    return f'Dataset convert to TSV {dataset_name}, version {version}: '
+
+
+def main():
+    parser = argparse.ArgumentParser('Convert a dataset to TSV(s)')
+    add_args_to_locate_dataset(parser)
+    parser.add_argument('--format', '-fm', type=enum_type(LineFormat), default=LineFormat.JSONL, help='Format of output data.', choices=list(LineFormat), required=False)
+    parser.add_argument('--flatten', '-fl', action='store_true', help="If an image has multiple annotations, one image will be flattend in to multiple entries with image being deuplicated.")
```
Comment (msxingpeng) [inline vision_datasets/commands/converter_to_line_oriented_format.py]: typo: duplicated

File: vision_datasets/commands/converter_to_line_oriented_format.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    JSONL = 'jsonl'
+    TSV = 'tsv'
+
+
+def logging_prefix(dataset_name, version):
+    return f'Dataset convert to TSV {dataset_name}, version {version}: '
+
+
+def main():
+    parser = argparse.ArgumentParser('Convert a dataset to TSV(s)')
```
Comment (msxingpeng) [inline vision_datasets/commands/converter_to_line_oriented_format.py]: to TSV or jsonl.

File: vision_datasets/commands/converter_to_line_oriented_format.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+logger = set_up_cmd_logger(__name__)
+
+
+class LineFormat(Enum):
+    JSONL = 'jsonl'
+    TSV = 'tsv'
+
+
+def logging_prefix(dataset_name, version):
+    return f'Dataset convert to TSV {dataset_name}, version {version}: '
```
Comment (msxingpeng) [inline vision_datasets/commands/converter_to_line_oriented_format.py]: Dataset converted to [one of LineFormat]

**Comment by gegeo0** [comment]
LGTM

**Pull request merged**